syntax = "proto3";

message getReq {
	string key = 1;
}

message getResp {
    int32 status = 1;
    string value = 2;
}

message putReq {
	string key = 1;
	string value = 2;
}

message putResp {
    int32 status = 1;
    string old_value = 2;
}

service kv_store {
	rpc get (getReq) returns (getResp) {}
	rpc put (putReq) returns (putResp) {}
}
